pandas/tests/indexes/datetimes/test_date_range.py

```diff
@@ -1166,3 +1166,50 @@ def test_date_range_with_custom_holidays():
         freq=freq,
     )
     tm.assert_index_equal(result, expected)
+
+
+class TestDateRangeNonNano:
+    def test_date_range_reso_validation(self):
+        msg = "'unit' must be one of 's', 'ms', 'us', 'ns'"
+        with pytest.raises(ValueError, match=msg):
+            date_range("2016-01-01", "2016-03-04", periods=3, unit="h")
+
+    def test_date_range_freq_higher_than_reso(self):
+        # freq being higher-resolution than reso is a problem
+        msg = "Use a lower freq or a higher unit instead"
+        with pytest.raises(ValueError, match=msg):
+            #    # TODO give a more useful or informative message?
+            date_range("2016-01-01", "2016-01-02", freq="ns", unit="ms")
+
+        # But matching reso is OK
+        date_range("2016-01-01", "2016-01-01 00:00:01", freq="ms", unit="ms")
```

Comment:
Could you assert the result of these calls? (and for these tests make a separate test function)?

Reply:
updated+green